gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -245,18 +247,20 @@ public void apply(JobListener jobListener, JobContext jobContext)
     final List<WorkUnit> skippedWorkUnits = new ArrayList<>();
     for (WorkUnit workUnit : workUnits.get()) {
       if (workUnit instanceof MultiWorkUnit) {
```

Comment:
Would it be possible to push the removeSkippedWorkUnits() method down to the WorkUnit, and the MultiWorkUnit classes, so that this code can read:
```java
for (WorkUnit wu: workUnits.get()) {
    wu.removeSkippedWorkUnits();
}
```
or somewhat similar? and then the particular logic to skip can move into the WorkUnit classes?